src/components/UserContainer.tsx

```diff
@@ -1,48 +1,68 @@
-}: UserContainerProps) => {
-    const currentRoomNoAction = !accepted && !requestSent && currentRoom;
-    let status = <p></p>;
-    if (accepted) {
-        status = <p className="user-display-status">Accepted</p>;
-    } else {
-        status = requestSent ? <p className="user-display-status">Pending</p> : <p></p>;
-    }
+type UserContainerState = {
+    hovered: boolean;
+}
 
-    return (
-        <div
-            className={`user-container ${currentRoom && 'user-container-current'} ${currentRoomNoAction &&
-                'user-container-initial'}`}
-            onMouseDown={(): void => {
-                onClick(displayName);
-            }}
-        >
-            <div
-                className={`user-display-icon ${currentRoomNoAction && 'user-display-light'}`}
-                style={{ backgroundColor: displayName.color }}
-            ></div>
-            <p className={`user-display-text ${currentRoomNoAction && 'user-display-light'}`}>
-                {displayName.displayName}
-            </p>
-            {status}
-        </div>
-    );
-};
+export default class UserContainer extends Component<UserContainerProps, UserContainerState> {
+    state: UserContainerState = {
+        hovered: false
+    };
 
-export default UserContainer;
+    render(): React.ReactNode {
+        const currentRoomNoAction = !this.props.accepted && !this.props.requestSent && this.props.currentRoom;
+        let status = <p></p>;
+        if (this.props.accepted) {
+            status = <p className="user-display-status">Accepted</p>;
+        } else {
+            status = this.props.requestSent ? <p className="user-display-status">Pending</p> : <p></p>;
+        }
+
+        return (
+            <div
+                className={`user-container ${this.props.currentRoom && 'user-container-current'} ${currentRoomNoAction &&
+                    'user-container-initial'}`}
+                onMouseDown={(): void => {
+                    this.props.onClick && this.props.onClick();
+                }}
+                onMouseEnter={(): void => {
+                    this.setState({ hovered: true });
+                }}
+                onMouseLeave={(): void => {
+                    this.setState({ hovered: false });
+                }}
+            >
+                <div
+                    className={`user-display-icon ${currentRoomNoAction && 'user-display-light'}`}
+                    style={{ backgroundColor: this.props.displayName.color }}
+                />
+                <p className={`user-display-text ${currentRoomNoAction && 'user-display-light'}`}>
+                    {this.props.displayName.displayName}
+                </p>
+                {
+                    this.state.hovered && this.props.onLeaveRoom ? (
+                        <div 
+                            className={'user-container-x-container'}
+                            onClick={(): void => { this.props.onLeaveRoom && this.props.onLeaveRoom(); }}
```

Comment:
And why is onLeaveRoom optional too?

Reply:
Only user containers that are under "Connected Rooms" should have the option to LeaveRoom. Again, I could create a separate component for this, but then we would be repeating CSS and most of what is rendered. I honestly don't know what the better way to go about this is.

Reply:
Well the click should be disabled if the user isn't connected right? So we can still pass the same `onLeaveRoom` function prop, and it just won't be able to be called until the user is connected?